Self-type doesn't seem to work for properties

I found this when trying to use self-type for pathlib (see PR python/typeshed#658). There are a few properties that have type PurePath and are repeated in the Path subclass. The stub uses plain attributes for these, but the implementation has properties, so I want to replace them with properties. However this doesn't seem to work. In particular I'm trying to add this to class PurePath:
```
    @property
    def parents(self: _P) -> Sequence[_P]: ...
    @property
    def parent(self: _P) -> _P: ...
```
Now I have an example class:
```
class MyPath(pathlib.PosixPath):
    def parity(self):
        return len(str(self)) % 2

root = MyPath('/')
reveal_type(root.parent)

```
This reveals MyPath as the type but *also* gives an error:
```
__tmp__.py:14: error: Revealed type is '__tmp__.MyPath*'
__tmp__.py:14: error: Invalid method type
```
If I remove the self-type, e.g.
```
    @property
    def parents(self) -> Sequence[PurePath]: ...
    @property
    def parent(self) -> PurePath: ...
```
the error goes away but the revealed type reverts back to (of course) PurePath.

@elazarg thoughts?

--- a/mypy/checkmember.py
+++ b/mypy/checkmember.py
@@ -323,7 +323,7 @@ def check_method_type(functype: FunctionLike, itype: Instance, is_classmethod: b
             # we passed to typ.fallback in analyze_member_access. See #1432.
             if isinstance(selfarg, TupleType):
                 selfarg = selfarg.fallback
-            if not subtypes.is_equivalent(selfarg, itype):
+            if not subtypes.is_subtype(selfarg, itype):
                 msg.invalid_method_type(item, context)
         else:
             # Check that cls argument has type 'Any' or valid class type.

--- a/test-data/unit/check-selftype.test
+++ b/test-data/unit/check-selftype.test
@@ -333,3 +333,21 @@ class E:
     def __new__(cls) -> E:
         reveal_type(cls)  # E: Revealed type is 'def () -> __main__.E'
         return cls()
+
+[case testSelfTypeProperty]
+from typing import TypeVar
+
+T = TypeVar('T', bound='A')
+
+class A:
+    @property
+    def member(self: T) -> T:
+        pass
+
+class B(A):
+    pass
+
+reveal_type(A().member)  # E: Revealed type is '__main__.A*'
+reveal_type(B().member)  # E: Revealed type is '__main__.B*'
+
+[builtins fixtures/property.pyi]

